### ISSUE TITLE
unable to view add data / choose json data (cant view drop down to select json data)

### ISSUE THREAD

**Issue by keshav1git** [WorldofDataScience/machine_flow#2]
<img width="947" alt="capture" src="https://user-images.githubusercontent.com/25166512/50506166-92c46b00-0a9d-11e9-8b58-269df55ef2e9.PNG">

please do look have , cant see any drop down to select json data which is in data folder

**Comment by sean-mcclure** [comment]
You have to **run the connect.js file inside the r_ml directory**.  MachineFlow requires you run the R service for any R code, and the Python Service for any Python code. So for your problem, _do the following_: 

First, **change directories into the r_ml folder** `cd r_ml/`

Then, **run the connect.js file using node** `node connect.js `

You should see the following in your terminal:

`R Machine Learning on port: 9191`

**Comment by keshav1git** [comment]
1) moved to python_ml directory and ran node connect.js
2) moved to root and started python server by using python -m http.server
3) added list_data api end point 
4) while adding the node got below error
<img width="953" alt="capture22" src="https://user-images.githubusercontent.com/25166512/50540967-f647b900-0bc1-11e9-89ed-12f821d7f3e2.PNG">

**Comment by keshav1git** [comment]
![image](https://user-images.githubusercontent.com/25166512/50540986-58a0b980-0bc2-11e9-8d8e-709ee4cabe20.png)

**Comment by sean-mcclure** [comment]
The list_data function is an R function. You must move into the r_ml
folder and run its connect.js file. There is a connect.js file for both
Python and R, and they sit in each of their respective folders. Make sure
you are inside the folder when running node connect.js.

**Comment by keshav1git** [comment]
1) ran python ml server running on port :8181 using node connect.js command
2) started python server using python -m http.server to serve machine_flow.html
3) created boston.json and placed in data folder
4) added a new api end point serving at port 8181 with function name list_data
5) tried to find boston.json under drop down but cant see.
<img width="960" alt="dropdown_missing" src="https://user-images.githubusercontent.com/25166512/50541104-5fc8c700-0bc4-11e9-9e9b-909f9ec776a2.png">

**Comment by sean-mcclure** [comment]
Let me know if my above comment worked.

**Comment by keshav1git** [comment]
No, i wasnt able to create a machine flow using python ml server
can you please elaborate more on how to create a flow.
steps i followed are as follows :
1) went to to python_ml folder and ran node connect.js on cmd 
--> D:\projects\machine flow\machine_flow\python_ml>node connect.js
Python Machine Learning on port: 8181


2) went to machine_flow root folder and ran python -m http.server
-->D:\projects\machine flow\machine_flow>python -m http.server
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...

please provide my further steps to continue

**Comment by keshav1git** [comment]
going through multiple errors
![image](https://user-images.githubusercontent.com/25166512/50591281-c72c9280-0eb4-11e9-9b17-e923dc5a8ab6.png)

**Comment by sean-mcclure** [comment]
As I said, you must go to r_ml server. You are only running the python, which is why it’s not working. Please see me previous answer. 

Sean

**Comment by sean-mcclure** [comment]
You are not running the connect.js file in the r_ml folder. Please refer to my answer above and read the README thoroughly.
